src/NuGet.Services.BasicSearch/SecretReader/SecretReaderFactory.cs

```diff
@@ -2,25 +2,26 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using NuGet.Indexing;
+using System.Security.Cryptography.X509Certificates;
 using NuGet.Services.KeyVault;
 
-namespace NuGet.Services.BasicSearch
+namespace NuGet.Services.BasicSearch.SecretReader
 {
-    internal class SecretReaderFactory : ISecretReaderFactory
+    public class SecretReaderFactory : ISecretReaderFactory
```

Comment:
Read about this: https://msdn.microsoft.com/en-us/library/system.runtime.compilerservices.internalsvisibletoattribute(v=vs.110).aspx

This is the correct approach to make stuff visible to test code